backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/tutor/tournament/TournamentService.java

```diff
@@ -0,0 +1,325 @@
+    @Transactional(isolation = Isolation.REPEATABLE_READ)
+    public void removeTournament(Integer userId, Integer tournamentId) {
+        User user = checkUser(userId);
+        Tournament tournament = checkTournament(tournamentId);
+
+        tournament.checkCreator(user);
+
+        Integer numberOfAnswers = 0;
+        if (tournament.hasQuiz())
+            numberOfAnswers = quizService.getQuizAnswers(tournament.getQuizId()).getQuizAnswers().size();
+        tournament.checkCanChange(numberOfAnswers);
+
+        tournament.remove();
+        tournamentRepository.delete(tournament);
+    }
+
+    @Retryable(value = { SQLException.class }, backoff = @Backoff(delay = 5000))
+    @Transactional(isolation = Isolation.REPEATABLE_READ)
+    public List<StudentDto> getTournamentParticipants(TournamentDto tournamentDto) {
+        Tournament tournament = checkTournament(tournamentDto.getId());
+
+        return tournament.getParticipants().stream().map(StudentDto::new).collect(Collectors.toList());
+    }
+
+    private void checkInput(Integer userId, Set<Integer> topicsId, TournamentDto tournamentDto) {
+        if (userId == null) {
+            throw new TutorException(TOURNAMENT_MISSING_USER);
+        }
+        if (topicsId == null) {
+            throw new TutorException(TOURNAMENT_MISSING_TOPICS);
+        }
+        if (tournamentDto.getStartTime() == null) {
+            throw new TutorException(TOURNAMENT_MISSING_START_TIME);
+        }
+        if (tournamentDto.getEndTime() == null) {
+            throw new TutorException(TOURNAMENT_MISSING_END_TIME);
+        }
+        if (tournamentDto.getNumberOfQuestions() == null) {
+            throw new TutorException(TOURNAMENT_MISSING_NUMBER_OF_QUESTIONS);
+        }
+    }
+
+    private User checkUser(Integer userId) {
+        User user = userRepository.findById(userId).orElseThrow(() -> new TutorException(USER_NOT_FOUND, userId));
+
+        if (user.getRole() != User.Role.STUDENT) {
+            throw new TutorException(USER_NOT_STUDENT, user.getId());
+        }
+
+        return user;
+    }
+
+    private Tournament checkTournament(Integer tournamentId) {
+        return tournamentRepository.findById(tournamentId)
+                .orElseThrow(() -> new TutorException(TOURNAMENT_NOT_FOUND, tournamentId));
+    }
+
+    private void createQuiz(Tournament tournament) {
+        StatementTournamentCreationDto quizForm = new StatementTournamentCreationDto();
+        quizForm.setNumberOfQuestions(tournament.getNumberOfQuestions());
+        quizForm.setTopicConjunction(tournament.getTopicConjunction());
+
+        StatementQuizDto statementQuizDto = statementService.generateTournamentQuiz(tournament.getCreator().getId(),
+                tournament.getCourseExecution().getId(), quizForm);
+
+        Quiz quiz = quizRepository.findById(statementQuizDto.getId())
+                .orElseThrow(() -> new TutorException(QUIZ_NOT_FOUND, statementQuizDto.getId()));
+
```

Comment:
Why the next statements are not done when it is generated?

Reply:
Done